python_modules/libraries/dagster-dbt/dagster_dbt/dbt_project.py

```diff
@@ -208,3 +216,8 @@ def sync_project_to_packaged_dir(
             ),
         )
         logger.log(log_level, "Sync complete.")
+
+    def manage_state_artifacts(self, project: DbtProject):
+        """Invoked when state_path is set on the project, override in a subclass
+        to customize behavior.
+        """
```

Comment:
raise an exception if `state_dir` is provided, but this method isn't overridden?

Reply:
the way things are currently factored one could set `state_dir` on `DbtProject` but not use `DagsterDbtProjectManager`  or only only use pieces of it so I lean away from raising

Reply:
Got it -- maybe include a logging statement, to signal that something ran when `state_dir` was provided? 

If the user wants to remove the log, they would have to override `manage_state_artifacts` with an empty method. That way, we have explicit opt-in for the folks who are doing this behavior of providing `state_dir` but don't want to do anything with `manage_state_artifacts`